docs/do-not-merge.md

```diff
@@ -0,0 +1,815 @@
+The `arrayIngestMode` context parameter controls how ARRAY type values are stored in Druid segments.
+
+When you set `arrayIngestMode` to `array` (recommended for SQL compliance), the MSQ task engine stores all ARRAY typed values in [ARRAY typed columns](https://druid.apache.org/docs/latest/querying/arrays) and supports storing both VARCHAR and numeric typed arrays.
+
+For backwards compatibility, `arrayIngestMode` defaults to `mvd`. When `"arrayIngestMode":"mvd"`, Druid only supports VARCHAR typed arrays and stores them as [multi-value string columns](https://druid.apache.org/docs/latest/querying/multi-value-dimensions).
+
+When you set `arrayIngestMode` to `none`, Druid throws an exception when trying to store any type of arrays.
+
+For more information on how to ingest `ARRAY` typed columns with SQL-based ingestion, see [SQL data types](https://druid.apache.org/docs/latest/querying/sql-data-types#arrays) and [Array columns](https://druid.apache.org/docs/latest/querying/arrays).
+
+## Incompatible changes
+
+### Removed Hadoop 2
+
+Support for Hadoop 2 has been removed.
+Migrate to SQL-based ingestion or JSON-based batch ingestion if you are using Hadoop 2.x for ingestion today.
+If migrating to Druid's built-in ingestion is not possible, you must upgrade your Hadoop infrastructure to 3.x+ before upgrading to Druid 28.0.0.
+
+[#14763](https://github.com/apache/druid/pull/14763)
+
+### Removed GroupBy v1 
+
+The GroupBy v1 engine has been removed. Use the GroupBy v2 engine instead.  
+If you are using GroupBy v1 in JSON-based batch queries, you must change your queries before upgrading. Otherwise, your queries will fail.
+For more information, see [GroupBy queries](https://druid.apache.org/docs/latest/querying/groupbyquery.html).
+
+[#14866](https://github.com/apache/druid/pull/14866)
+
+### Removed Coordinator dynamic configs
+
+The `decommissioningMaxPercentOfMaxSegmentsToMove` config has been removed.
+The use case for this config is handled by smart segment loading now, which is enabled by default.
+
+[#14923](https://github.com/apache/druid/pull/14923)
+
+### Removed `cachingCost` strategy
+
+The `cachingCost` strategy for segment loading has been removed.
+Use `cost` instead, which has the same benefits as `cachingCost`.
+
+If you have `cachingCost` set, the system ignores this setting and automatically uses `cost`.
+
+[#14798](https://github.com/apache/druid/pull/14798)
+
+### Removed `InsertCannotOrderByDescending`
+
+The deprecated MSQ fault `InsertCannotOrderByDescending` has been removed.
+
+[#14588](https://github.com/apache/druid/pull/14588)
+
+### Removed the backward compatibility code for the Handoff API
+
+The backward compatibility code for the Handoff API in `CoordinatorBasedSegmentHandoffNotifier` has been removed.
+
+[#14652](https://github.com/apache/druid/pull/14652)
+
```

Comment:
heh, should we call out if you are upgrading from a Druid version older than 0.14 you're going to have a bad time? (probably not necessary, but it might be nicer to make it super clear that this is a very old backwards compat thing so people don't worry)

Reply:
@clintropolis If we do add a note about upgrading from a Druid version older than 0.14, what should the message say? What are we warning the users about?

Reply:
I'm not sure we even need to mention it? Otherwise, i guess we could mention something about upgrading to a newer version of Druid before upgrading to 28

Reply:
Added a note about upgrading from Druid 0.14.0